src/lib/FormGroupLayout.tsx

```diff
@@ -1,37 +1,63 @@
-import { PropsWithChildren } from "react";
+import { PropsWithChildren, useState } from "react";
 import { FieldValues, useFormContext } from "react-hook-form";
-import { FormGroup, Label, FormFeedback, FormText } from "reactstrap";
+import { FormGroup, Label, FormFeedback, FormText, Tooltip } from "reactstrap";
 import { useSafeNameId } from "src/lib/hooks/useSafeNameId";
 import { CommonInputProps } from "./types/CommonInputProps";
+import "./styles/FormGroupLayout.css"
 
 interface FormGroupLayoutProps<T extends FieldValues>
-  extends PropsWithChildren<Pick<CommonInputProps<T>, "helpText" | "label" | "name" | "id">> {
-  layout?: "checkbox" | "switch";
+    extends PropsWithChildren<Pick<CommonInputProps<T>, "helpText" | "label" | "name" | "id" | "labelToolTip">> {
+    layout?: "checkbox" | "switch";
 }
 
 const FormGroupLayout = <T extends FieldValues>(props: FormGroupLayoutProps<T>) => {
-  const { label, helpText, children, layout } = props;
-  const { name, id } = useSafeNameId(props.name, props.id);
-  const {
-    formState: { errors },
-  } = useFormContext();
+    const { label, helpText, children, layout, labelToolTip } = props;
+    const { name, id } = useSafeNameId(props.name, props.id);
+    const {
+        formState: { errors },
+    } = useFormContext();
 
-  const fieldError = errors[name];
-  const errorMessage = String(fieldError?.message);
+    const fieldError = errors[name];
+    const errorMessage = String(fieldError?.message);
 
-  const switchLayout = layout === "switch";
-  const checkboxLayout = layout === "checkbox";
+    const switchLayout = layout === "switch";
+    const checkboxLayout = layout === "checkbox";
 
-  return (
-    <FormGroup switch={switchLayout ? true : undefined} check={checkboxLayout ? true : undefined}>
-      <Label check={checkboxLayout || switchLayout} for={id}>
-        {label}
-      </Label>
-      {children}
-      <FormFeedback>{errorMessage}</FormFeedback>
-      {helpText && <FormText>{helpText}</FormText>}
-    </FormGroup>
-  );
+    const [tooltipOpen, setTooltipOpen] = useState(false);
+    const toggle = () => setTooltipOpen(!tooltipOpen);
```

Comment:
this will work in 99% of the cases but is not best practice. 
When you want to set a state that is dependent on the previous state it's better to supply a function to the the setState method like so:

`    const toggle = () => setTooltipOpen((prev) => !prev);`

Reply:
ref: https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
Switched to UncontrolledTooltip. This is now obsolete and I removed the whole thing